icu4j/main/core/src/main/java/com/ibm/icu/text/NFRule.java

```diff
@@ -303,18 +301,18 @@ private String parseRuleDescriptor(String description) {
             if (firstChar >= '0' && firstChar <= '9' && lastChar != 'x') {
                 // if the rule descriptor begins with a digit, it's a descriptor
                 // for a normal rule
-                long tempValue = 0;
+                long val = 0;
```

Comment:
I did some syncing between ICU4C and ICU4J. Changes like this had to be crossed over so that it's easier to compare.